HW04.Rmd

```diff
@@ -1,126 +1,239 @@
+
+Assume that:
+
+* Winter = January-March
+* Spring = April-June
+* Summer = July-September
+* Fall = October-December
+
+```{r season}
+gun_deaths_top_season <- fct_collapse(month.abb[gun_deaths$month],
+  Winter = c("Jan", "Feb", "Mar"),
+  Spring = c("Apr", "May", "Jun"),
+  Summer = c("Jul", "Aug", "Sep"),
+  Fall = c("Oct", "Nov", "Dec")) %>%
+  fct_count() %>%
+  top_n(1, n)
+
+paste("Season with the most gun deaths:", gun_deaths_top_season[[1]])
+```
+
+
+# These are more open ended questions from here on out, you could look at making multiple summaries or types of plots. 
+
+### Are whites who are killed by guns more likely to die because of suicide or homicide? How does this compare to blacks and hispanics?
+
+```{r race}
+# Would nont have been able to figure out how to put percentages on the graphs without this tutorial:
+# https://rkabacoff.github.io/datavis/Bivariate.html
+
+gun_deaths_by_intent_and_race <- gun_deaths %>%
+  group_by(race, intent) %>%
+  summarize(n = n()) %>% 
+  mutate(pct = n/sum(n),
+         lbl = scales::percent(pct))
+gun_deaths_by_intent_and_race
+
+ggplot(gun_deaths_by_intent_and_race, aes(race, y = pct, fill = intent)) +
+  geom_bar(stat = "identity", position = "fill") +
+  geom_text(aes(label = lbl), size = 3, position = position_stack(vjust = 0.5)) +
+  theme(axis.text.x = element_text(vjust = 0.5, hjust=0.5)) +
+  scale_y_continuous(breaks = seq(0, 1, .2), label = percent) +
+  scale_fill_brewer(palette = "Set3", direction=1) +
+  xlab("Race") +
+  ylab("Percent of Deaths") +
+  labs(fill = "Intent")
+```
+
+(hint maybe looking at percentages would be good)
+
+Answer: Whites who are killed by guns are more likely to die of suicide than homicide. Black and hispanic people killed by guns are more likely to die of homicide.
+
+### Are police-involved gun deaths significantly different from other gun deaths? Assess the relationship between police involvement and age, police involvement and race, and the intersection of all three variables.
+
+```{r police and age}
+ggplot(gun_deaths, aes(factor(police, levels = c(0, 1), labels = c("No", "Yes")), age)) +
+  geom_boxplot() +
+  xlab("Police Involvement") +
+  ylab("Age") +
+  coord_flip()
+```
+
+```{r police and race}
+gun_deaths_by_police_and_race <- gun_deaths %>%
+  group_by(race, police) %>%
+  summarize(n = n()) %>% 
+  mutate(pct = n/sum(n),
+         lbl = scales::percent(pct))
+
+ggplot(gun_deaths_by_police_and_race, 
+       aes(race, y = pct, fill = factor(police, levels = c(0, 1), labels = c("No", "Yes")))) +
+  geom_bar(stat = "identity", position = "fill") +
+  geom_text(aes(label = lbl), size = 3, position = position_stack(vjust = 0.5)) +
+  theme(axis.text.x = element_text(vjust = 0.5, hjust=0.5)) +
+  scale_y_continuous(breaks = seq(0, 1, .2), label = percent) +
+  scale_fill_brewer(palette = "Set2") +
+  xlab("Race") +
+  ylab("Percent of Deaths") +
+  labs(fill = "Police\nInvolvement")
+```
```

Comment:
The graphs look great. I also see the Native American/Native Alaskan look right in the graph x-axis. So disregard my prior comment. You got farther along than me. I did not make it to the police analysis.